core/models/asset.py

```diff
@@ -359,6 +359,9 @@ def clean(self):
 
     def save(self, *args, **kwargs):
         self.full_clean()
+        log = AssetLog.objects.filter(asset=self.asset).first()
+        if log and log.log_type == self.log_type:
+            return None
```

Comment:
let's raise an exception here instead of returning `None`. (`ValidationError` should do). Then we should also update the API so that it will be clear in the response that the asset has already been checked in or out

Reply:
I've returned None on the model level and raise a ValidationError at the serializer level